nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AttributesToCSV.java

```diff
@@ -0,0 +1,272 @@
+        "attribute value is returned unchanged.")
+@WritesAttribute(attribute = "CSVAttributes", description = "CSV representation of Attributes")
+public class AttributesToCSV extends AbstractProcessor {
+
+    private static final String OUTPUT_NEW_ATTRIBUTE = "flowfile-attribute";
+    private static final String OUTPUT_OVERWRITE_CONTENT = "flowfile-content";
+    private static final String OUTPUT_ATTRIBUTE_NAME = "CSVAttributes";
+    private static final String OUTPUT_SEPARATOR = ",";
+    private static final String OUTPUT_MIME_TYPE = "text/csv";
+
+
+    public static final PropertyDescriptor ATTRIBUTES_LIST = new PropertyDescriptor.Builder()
+            .name("attribute-list")
+            .displayName("Attribute List")
+            .description("Comma separated list of attributes to be included in the resulting CSV. If this value " +
+                    "is left empty then all existing Attributes will be included. This list of attributes is " +
+                    "case sensitive and does not support attribute names that contain commas. If an attribute specified in the list is not found it will be emitted " +
+                    "to the resulting CSV with an empty string or null depending on the 'Null Value' property. " +
+                    "If a core attribute is specified in this list " +
+                    "and the 'Include Core Attributes' property is false, the core attribute will be included. The attribute list " +
+                    "ALWAYS wins.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor DESTINATION = new PropertyDescriptor.Builder()
+            .name("destination")
+            .displayName("Destination")
+            .description("Control if CSV value is written as a new flowfile attribute 'CSVAttributes' " +
+                    "or written in the flowfile content. Writing to flowfile content will overwrite any " +
+                    "existing flowfile content.")
+            .required(true)
+            .allowableValues(OUTPUT_NEW_ATTRIBUTE, OUTPUT_OVERWRITE_CONTENT)
+            .defaultValue(OUTPUT_NEW_ATTRIBUTE)
+            .build();
+
+    public static final PropertyDescriptor INCLUDE_CORE_ATTRIBUTES = new PropertyDescriptor.Builder()
+            .name("include-core-userSpecifiedAttributes")
+            .displayName("Include Core Attributes")
+            .description("Determines if the FlowFile org.apache.nifi.flowfile.attributes.CoreAttributes, which are " +
+                    "contained in every FlowFile, should be included in the final CSV value generated.  The Attribute List property " +
+                    "overrides this setting.")
+            .required(true)
+            .allowableValues("true", "false")
```

Comment:
Should add StandardValidators.BOOLEAN_VALIDATOR

Reply:
done